crates/miden-tx/src/executor/mod.rs

```diff
@@ -360,31 +360,41 @@ where
         .map_err(TransactionExecutorError::TransactionProgramExecutionFailed);
 
         match result {
-            Ok(_) => Ok(NoteAccountExecution::Success),
-            Err(tx_execution_error) => {
+            Ok(_) => Ok(NoteConsumptionInfo::new_successful(
+                tx_inputs
+                    .into_input_notes()
+                    .into_iter()
+                    .map(|note| note.into_note())
+                    .collect::<Vec<_>>(),
+            )),
+            Err(error) => {
+                // Map the last note id from execution to the failed note.
                 let notes = host.tx_progress().note_execution();
-
-                // empty notes vector means that we didn't process the notes, so an error
-                // occurred somewhere else
-                if notes.is_empty() {
-                    return Err(tx_execution_error);
-                }
```

Comment:
Do we no longer need to handle this case?

Btw, do we check anywhere if the set of input notes is empty? We should make sure that if, for w/e reason this function is called with no notes, it returns a meaningful result (e.g., an empty `NoteConsumptionInfo`).